Apps/Database/src/Delegates/DBCommunicationDelegate.cs

```diff
@@ -181,5 +216,51 @@ public DBResult<Communication> Delete(Communication communication, bool commit =
             this.logger.LogDebug($"Finished deleting Communication in DB");
             return result;
         }
+
+        private List<Communication> GetActiveBannerCommunications()
+        {
+            List<Communication> communication = this.dbContext.Communication
+                 .OrderByDescending(c => c.CreatedDateTime)
+                 .Where(c => c.CommunicationTypeCode == CommunicationType.Banner
+                     && c.CommunicationStatusCode == CommunicationStatus.New && DateTime.UtcNow <= c.ExpiryDateTime)
+                 .ToList();
+
+            return communication;
+        }
+
+        private bool IsCommunicationValidCommit(Communication communication)
+        {
+            List<Communication> activeBannerCommunication = this.GetActiveBannerCommunications();
+
+            if (communication.CommunicationStatusCode == CommunicationStatus.New)
+            {
+                if (activeBannerCommunication != null && activeBannerCommunication.Count == 0)
+                {
+                    return true;
+                }
+                else
+                {
+                    var validCommunications = activeBannerCommunication
+                            .OrderByDescending(c => c.EffectiveDateTime)
+                            .Where(c => (c.ExpiryDateTime < communication.ExpiryDateTime &&
+                                        communication.EffectiveDateTime > c.ExpiryDateTime) ||
+                                        (communication.EffectiveDateTime < c.EffectiveDateTime && communication.ExpiryDateTime < c.EffectiveDateTime))
+                            .ToList();
+
+                    if (validCommunications != null && validCommunications.Count > 0)
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        return false;
+                    }
```

Comment:
Nit: Sonar analysis will say to remove the else and just return false.

Example:

`If (condition)
 {
      return true;
 }
return false;`